adapter/src/main/kotlin/org/javacs/ktda/jdi/launch/JDILauncher.kt

```diff
@@ -75,12 +75,16 @@ class JDILauncher(
 		.let { it.find { it.javaClass.name == "com.sun.tools.jdi.SunCommandLineLauncher" } ?: it.firstOrNull() }
 		?: throw KotlinDAException("Could not find a launching connector (for a new debuggee VM)")
 	
-	private fun sourcesRootsOf(projectRoot: Path): Set<Path> = projectRoot.resolve("src")
-		.let(Files::list) // main, test
-		.filter { Files.isDirectory(it) }
-		.flatMap(Files::list) // kotlin, java
-		.filter { Files.isDirectory(it) }
-		.collect(Collectors.toSet())
+	private fun sourcesRootsOf(projectRoot: Path): Set<Path> =
+			Files.walk(projectRoot, 2) // root project and submodule
```

Comment:
Hm, ideally I'd like to not hardcode the 2 levels, since there are lots of edge cases where this wouldn't work (project in subfolder, nested subprojects, etc.). The current logic is already somewhat hacky and in the medium-term having a clean solution would almost certainly pay off, in the sense that we shouldn't have to account for all of these scenarios manually

Reply:
What we should probably do is query Maven/Gradle to find source files and fall back to walking the source tree only if everything else fails

Reply:
I agree with you on the ideally part here. A Maven/Gradle query seems like a much better long term solution 🙂 
 Development velocity is not super fast in this project, so might take some time. We have several people who want to use the project to debug multi-module codebases, which are atm not possible. Merging this PR might not be perfect, but it moves us slowly a step forward.


**My suggestion:** We make an issue of your suggestion for a more permanent fix, with a link to the current solution (just a permalink to the code changed in this PR). In this issue we write what you have written above about a Maven/Gradle query and the fallback. Part of a fallback might also be an optional input with maximum source root depth or something? Anyway, we can put information like that in the issue-text. Does that sound okay to you @fwcd ? 🙂

Reply:
**I agree with @themkat 's suggestion.**

I knew the PR was not perfect at that time, but it can resolve the problem temporarily. I tried to query Maven/Gradle and classic Eclipse project source directories by parsing their's description file or using LSP information, but it is a bit complex.

I think the long-term solution have to query project's source construction,  but it need some time to do.

Reply:
Yeah I suppose supporting this use case for now is better than what we have. In any case, this snippet is in dire need of refactoring since it makes too many assumptions about the user's project layout, but let's move that to a future PR.

Reply:
Created an issue for this #77.